aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -203,5 +204,48 @@ Tensor& logical_xor_(Tensor& self, const Tensor& other) {
   return native::logical_xor_out(self, self, other);
 }
 
+Tensor& lt_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  TORCH_CHECK(result.scalar_type() == kBool,
+      "The output tensor of lt must be a bool.");
+  auto iter = TensorIterator::comparison_op(result, self, other, true);
+  lt_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor lt(const Tensor& self, const Tensor& other) {
+  Tensor result = at::empty({0}, self.options().dtype(kBool));
+  native::lt_out(result, self, other);
+  return result;
+}
+
+Tensor& lt_(Tensor& self, const Tensor& other) {
+  TORCH_CHECK(self.dtype() == other.dtype(),
+      "Expected object of scalar type ", self.dtype(), " but got scalar type ",
+      other.dtype(), "for argument 'other' in call to lt_")
+  auto iter = TensorIterator::comparison_op(self, self, other, true);
+  lt_stub(iter.device_type(), iter);
+  return self;
+}
+
+Tensor& lt_out(Tensor& result, const Tensor& self, Scalar other) {
+  AT_DISPATCH_ALL_TYPES_AND(at::ScalarType::Half, self.scalar_type(), "lt_out", [&]{
+    // Validate that is possible to convert scalar to tensor dtype without overflow
```

Comment:
@nairbv -- how does this look to you? I remember you saying that in other ops (add sub mul div) we truncate the Scalar because it doesn't get promoted.

Reply:
it's different for arithmetic ops. I added comment about how is currently implemented for logical ops.

Reply:
Sorry, where is the comment?

Reply:
https://github.com/pytorch/pytorch/pull/25998#issuecomment-534685473

Reply:
You should add a comment here into the code. By reading the code, I'm not sure why we've implemented it like this or that arithmetic ops have different behavior.

Reply:
But it's there, right? What kind of comment?

Reply:
"it's different for arithmetic ops. for arithmetic ops we don't verify this. The reason is because previous TH behavior checked for overflow in comparison ops but not arithmetic ops. In the future, we should reconsider this inconsistency and decide if we're going to check overflow for arithmetic ops"